### PR TITLE
Remove metric registry and fix examples.

### PR DESCRIPTION
**Pull request by bogdandrutu**
This makes Meter similar with Tracer and Tagger.

### CODE REVIEW

File: api/src/main/java/openconsensus/metrics/Meter.java [hunk 1]
```diff
@@ -16,13 +16,50 @@
 
 package openconsensus.metrics;
 
-/** Entry point fot metrics API, this object allows to create new {@link MetricRegistry}. */
+/** Entry point fot metrics API, this object allows to record measurements and {@link Metric}s. */
 public interface Meter {
 
   /**
-   * Returns a new builder for a {@code MetricRegistry}.
+   * Returns a builder for a {@link GaugeLong} to be added to the registry.
```
Comment (songy23) [inline api/src/main/java/openconsensus/metrics/Meter.java]: Nit: remove "to be added to the registry". Same for all other javadocs.

File: api/src/main/java/openconsensus/metrics/Metric.java [hunk 2]
```diff
@@ -118,6 +119,32 @@
...
+     * <p>The final name of the reported metric will be <code>component + "_" + name</code> if the
+     * component is not empty.
+     *
+     * <p>It is recommended to always set a component name for all the metrics, because some
+     * implementations may filter based on the component.
+     *
+     * @param component the name of the component that reports these metrics.
+     * @return this.
+     */
+    B setComponent(String component);
```
Comment (songy23) [inline api/src/main/java/openconsensus/metrics/Metric.java]: Nit: consider `setComponentName`

**Comment by bogdandrutu** [comment]
Closed and reopened to trigger travis :)

**Pull request merged**